spec/services/draw_activator_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+RSpec.describe DrawActivator do
+  # TODO: extract this into a shared example; if we're going to use this pattern
+  # across all of our service objects we should have a way to easily implement
+  # it for all of them. Should be pretty easy - all we need to pass to the
+  # shared example is the method we want to call and the params hash
+  describe '.start' do
+    it 'calls :start on an instance of DrawStarter' do
+      draw = instance_spy('Draw')
+      draw_activator = create_mock_draw_activator(draw: draw)
+      described_class.activate(draw: draw)
+
+      expect(draw_activator).to have_received(:activate)
+    end
+  end
+
+  describe '#activate' do
+    it 'checks to make sure that the draw is a draft' do
+      draw = instance_spy('draw', draft?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to include('Draw must be a draft.')
+    end
+
+    it 'checks to make sure that the draw has at least one student' do
+      draw = instance_spy('draw', has_students?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to \
+        include('Draw must have at least one student.')
+    end
+
+    it 'checks to make sure that there are enough beds for students' do
+      draw = instance_spy('draw', has_enough_beds?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to \
+        include('Draw must have at least one bed per student')
+    end
+
+    it 'updates the status of the draw to pre_lottery' do
+      # draw = create_mock_draw
+      draw = create_draw_data
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:obj]).to be_pre_lottery
+    end
```

Comment:
I was able to get the following to pass without persistence:

      draw = instance_spy('Draw', draft?: true, has_students?: true,
                                  has_enough_beds?: true)

      described_class.activate(draw: draw)

      expect(draw).to have_received(:update_attributes)
        .with(status: 'pre_lottery')

Which I think tests what we want it to?